index.js

```diff
@@ -37,6 +38,12 @@ const stripBearer = str =>
 const throwIfNull = val =>
   val == null ? Promise.reject('invalid token') : val
 
+const throwIfExpired = ({ ignoreExpiration }) => exp => {
+  if (!ignoreExpiration && exp && exp > (Date.now() / 1000)) {
+    Promise.reject(new TokenExpiredError('JWT expired', new Date(exp * 1000)))
+  }
+}
```

Comment:
The mixed return type of `Promise` or `undefined` is a bit smelly. This is a synchronous function, so maybe...

```suggestion
const throwIfExpired = ({ ignoreExpiration }) => exp => {
  if (!ignoreExpiration && exp && exp > (Date.now() / 1000)) {
    throw new TokenExpiredError('JWT expired', new Date(exp * 1000))
  }
}
```

Reply:
Yup, makes sense @mgreystone.